### PR TITLE
Issue #3089823: Delete action is not performed if not after change action

### PR DESCRIPTION
**Pull request by mtodor**
Problem is described here: https://www.drupal.org/project/update_helper/issues/3089823

PR includes the following:
- test for the described problem
- handling of applied configuration changes. First, all changes will be applied and then new configuration will be verified against active configuration and only if it's exactly same, it will be marked as update was already successfully applied.

### CODE REVIEW

File: src/ConfigHandler.php [hunk 1]
```diff
@@ -262,9 +262,9 @@ protected function getExpectedConfig(array $diffs) {
    */
   protected function getUpdateConfig(array $diffs) {
     $list_update = [
+      'delete' => [],
```
Comment (dbosen) [inline src/ConfigHandler.php]: Could you explain the order change? I don't understand it.

File: tests/src/Kernel/UpdaterTest.php [hunk 2]
```diff
@@ -131,6 +131,28 @@ protected function setUp() {
...
+    /** @var \Drupal\update_helper\ConfigHandler $config_handler */
+    $config_handler = \Drupal::service('update_helper.config_handler');
+
+    // Create update configuration for testExecuteUpdate.
+    $patch_file_path = $config_handler->getPatchFile('update_helper', 'test_updater', TRUE);
+    file_put_contents($patch_file_path, $yml_serializer->encode($this->getUpdateDefinition()));
+
+    // Create update configuration for testOnlyDeleteUpdate.
+    $patch_file_path = $config_handler->getPatchFile('update_helper', 'test_updater_only_delete', TRUE);
+    file_put_contents($patch_file_path, $yml_serializer->encode(
```
Comment (dbosen) [inline tests/src/Kernel/UpdaterTest.php]: same as above

File: tests/src/Kernel/UpdaterTest.php [hunk 3]
```diff
@@ -131,6 +131,28 @@ protected function setUp() {
     /** @var \Drupal\Core\Serialization\Yaml $yml_serializer */
     $yml_serializer = \Drupal::service('serialization.yaml');
     file_put_contents($this->configDir . '/install/tour.tour.tour-update-helper-test.yml', $yml_serializer->encode($tour_config));
+
+    /** @var \Drupal\update_helper\ConfigHandler $config_handler */
+    $config_handler = \Drupal::service('update_helper.config_handler');
+
+    // Create update configuration for testExecuteUpdate.
+    $patch_file_path = $config_handler->getPatchFile('update_helper', 'test_updater', TRUE);
+    file_put_contents($patch_file_path, $yml_serializer->encode($this->getUpdateDefinition()));
```
Comment (dbosen) [inline tests/src/Kernel/UpdaterTest.php]: encode() is a static function.

**Review by dbosen — CHANGES_REQUESTED** [review]

**Comment by mtodor** [comment]
Update Tests on Thunder 2
https://travis-ci.org/BurdaMagazinOrg/thunder-distribution/jobs/657793744?utm_medium=notification&utm_source=github_status

**Comment by mtodor** [comment]
Here is Test on Thunder 3:
https://travis-ci.com/thunder/thunder-distribution/jobs/293482614

**Review by dbosen — APPROVED** [review]
lgtm

**Pull request merged**